### PR TITLE
Glm4Moe: Switching moe routing type to ds3 seems to make inference work

### PR DESCRIPTION
**Pull request by MikeRoz47**
Trying to run GLM-4.5-Air with the current dev branch works, but trying to run GLM-4.5 (non-Air) does not. It produces repetition that looks like this (3.0bpw quant of GLM-4.5):

```
You are a helpful AI assistant.

User: What is two plus two?

Assistant: You are a helpful AI assistant.</think> What is one plus one.</think> You are a helpful AI assistant.</think> What is three plus three.</think> You are a helpful AI assistant.</think> You are a helpful AI assistant.</think>
You are a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a
helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI
assistant.</think> You is a helpful AI assistant.</think> You is a helpful AI assistant.</think> You is a

 !! Response exceeded 1000 tokens and was cut short.
```

Switching the `router_type` in `Glm4MoeModel` seems to make quants that were built with the existing dev code work:

```
You are a helpful AI assistant.

User: What is two plus two?

Assistant: <think>The user is asking a basic arithmetic question: "What is two plus two?"

This is a simple addition problem. The answer is 2 + 2 = 4.

I should provide a straightforward and helpful response to this basic question.</think>Two plus two equals four.

User:
```

What I haven't tested yet (but will kick off as soon as I submit this PR) is whether a quant made using this code can also be used for inference with this code.

### CODE REVIEW

**Comment by turboderp** [comment]
Thank you for this. The problem is actually with an assumption made in the routing kernel that the correction bias is always positive, so sigmoid(logits) + bias is positive, and so radix sorting the floating-point scores is valid. The larger GLM4.5 seems to have a negative bias in places, so the assumption breaks. I've pushed a fix to address it in the kernel.
